src/dfilemaker/dfilemaker.c

```diff
@@ -244,37 +247,58 @@ int getnum(const char* fname)
 }
 
 //-----------------------------------
-// put nwds random ints into buffer
+// put nwds ints into buffer
 //------------------------------------
-void fillbuff(int* ibuff, int nwds)
+void fillbuff(int* ibuff, size_t nwds, int kft)
 {
     int i;
-    for (i = 0; i < nwds; i++) {
-        ibuff[i] = rand();
+    switch (kft)
+    {
+        case 0:
+          for (i = 0; i < nwds; i++) {
+            ibuff[i] = rand();
+          }
+          break;
+        case 1:
+          for (i = 0; i < nwds; i++) {
+            ibuff[i] = 0xFFFFFFFF;
+          }
+          break;
+        case 2:
+          for (i = 0; i < nwds; i++) {
+            ibuff[i] = 0x00000000;
+          }
+          break;
+        case 3:
+          for (i = 0; i < nwds; i++) {
+            ibuff[i] = 0xAAAAAAAA;
+          }
+          break;
     }
 }
 
-size_t bufsize = 1024 * 1024;
+size_t bufsize = 1024 * 1024 * 4;
 char* buf;
 size_t size, isize;
 int nnum;
 
 /*----------------------------------------------*/
 /* add content to a node created by create_file */
 /*----------------------------------------------*/
-static int write_file(mfu_flist list, uint64_t idx)
+static int write_file(mfu_flist list, uint64_t idx, int kft)
 {
     int rc = 0;
 
     /* get destination name */
     const char* dest_path = mfu_flist_file_get_name(list, idx);
+
+    /* get destination size */
     uint64_t fsize =  mfu_flist_file_get_size(list, idx);
     size = fsize;
-    isize = (size + 1) / 2;
-    //printf("writing file %s, fsize = %li, size = %li\n",dest_path,fsize,size);
     nnum = getnum(dest_path);
     srand(nnum);
-    fillbuff((int*)buf, isize);
+    isize = bufsize/4;
```

Comment:
I don't see the +1/2 you're talking about; but the bufsiz/4 is because `fillbuf()` writes in terms of `int`s, which on x86_64 are 4 bytes long.  I'll clean that up so it's clearer and less prone to being accidentally broken.

Reply:
**isize = (size + 1) / 2;**

Reply:
I don't know why that was there.